src/main/java/bio/terra/workspace/service/resource/controlled/flight/delete/DeleteControlledResourceGcsBucketFlight.java

```diff
@@ -36,55 +42,115 @@ public DeleteControlledResourceGcsBucketFlight(FlightMap inputParameters, Object
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+
     // Flight plan:
-    // 1. TODO: Delete the Sam resource - this should make the object invisible via get and
-    // enumerate endpoints
+    // 1. Delete the Sam resource. That will make the object inaccessible.
     // 2. Delete the cloud resource:
     //    a. Set the lifecycle on the bucket to delete immediately
     //    b. Try deleting the bucket
     //    c. If delete succeeds, finish step
     //    d. If delete fails, sleep one hour; goto (either a or b; maybe a for belts and suspenders)
     // 3. Delete the metadata
+    addStep(
+        new DeleteSamResourceStep(
+            flightBeanBag.getResourceDao(),
+            flightBeanBag.getSamService(),
+            workspaceId,
+            resourceId,
+            userRequest));
     addStep(
         new DeleteGcsBucketStep(
             flightBeanBag.getCrlService(),
             flightBeanBag.getResourceDao(),
-            flightBeanBag.getWorkspaceService()));
-    addStep(new DeleteMetadataStep(flightBeanBag.getResourceDao()));
+            flightBeanBag.getWorkspaceService(),
+            workspaceId,
+            resourceId));
+    addStep(new DeleteMetadataStep(flightBeanBag.getResourceDao(), workspaceId, resourceId));
+  }
+
+  static class DeleteSamResourceStep implements Step {
+    private final ResourceDao resourceDao;
+    private final SamService samService;
+    private final UUID workspaceId;
+    private final UUID resourceId;
+    private final AuthenticatedUserRequest userRequest;
+
+    // TODO: this looks generic, so factor it out and share it
+    public DeleteSamResourceStep(
+        ResourceDao resourceDao,
+        SamService samService,
+        UUID workspaceId,
+        UUID resourceId,
+        AuthenticatedUserRequest userRequest) {
+      this.resourceDao = resourceDao;
+      this.samService = samService;
+      this.workspaceId = workspaceId;
+      this.resourceId = resourceId;
+      this.userRequest = userRequest;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext flightContext)
+        throws InterruptedException, RetryException {
+
+      WsmResource wsmResource = resourceDao.getResource(workspaceId, resourceId);
+      ControlledResource resource = wsmResource.castToControlledResource();
+
+      try {
+        samService.deleteControlledResource(resource, userRequest);
+      } catch (SamApiException samApiException) {
+        if (samApiException.getApiExceptionStatus() == HttpStatus.NOT_FOUND.value()) {
+          logger.debug("No Sam resource found for resource {}", resourceId);
+        } else {
+          throw samApiException;
+        }
+      }
+      return StepResult.getStepResultSuccess();
+    }
+
+    @Override
+    public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
+      // No undo for delete
+      return StepResult.getStepResultSuccess();
```

Comment:
Should this be a fatal error? I think it's technically possible that we delete the Sam resource and then for whatever reason start undoing - "success" implies we've returned to the previous state (i.e. restore the deleted Sam resource), which we can't. otoh I don't want to raise a fatal stairway response every time the Sam call fails

Reply:
It is a toss up. In TDR, we decided to just delete as much as we could and not error on the theory that it would help clean up corrupted state. The alternative is to cause a DISMAL_FAILURE (returning FATAL here) which, in production, should cause a human to take a look at what is going on.

Worth discussing further. Once we implement Sam retries, we should not be seeing transient Sam errors here.